src/hotspot/share/runtime/safepoint.cpp

```diff
@@ -67,6 +67,8 @@
 #include "utilities/macros.hpp"
 #include "utilities/systemMemoryBarrier.hpp"
 
+intptr_t safepointTimedOutThread = p2i(nullptr);
```

Comment:
Maybe not a global variable. 

I would either make this a static member of SafePoint and the other one a static member of Handshake, or (maybe simpler) both static members of VMError, plus accessors.